react/AppIcon/index.jsx

```diff
@@ -0,0 +1,73 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import classNames from 'classnames'
+import styles from './styles.styl'
+
+import Icon from '../Icon'
+
+import appDefaultIcon from '../../assets/icons/ui/icon-cube.svg'
+
+export class AppIcon extends Component {
+  constructor(props, context) {
+    super(props, context)
+    this.state = {
+      status: 'idle'
+    }
+  }
+
+  async componentDidMount() {
+    this.fetchIcon()
+  }
+
+  async fetchIcon() {
+    const { app, fetchIcon } = this.props
+
+    if (typeof this.props.fetchIcon !== 'function') {
+      throw new Error('fetchIcon prop is required')
+    }
```

Comment:
MINOR

I think this is already checked by the PropTypes. 

`fetchIcon` is destructured above so we can remove `this.props.`